src/mds/MDCache.cc

```diff
@@ -10735,6 +10809,12 @@ void MDCache::encode_replica_inode(CInode *in, mds_rank_t to, bufferlist& bl,
 
   in->_encode_base(bl, features);
   in->_encode_locks_state_for_replica(bl, mds->get_state() < MDSMap::STATE_ACTIVE);
+
+  if (struct_v >= 2) {
+    __u32 state = in->state;
+    encode(state, bl);
+  }
```

Comment:
Can it not be done conditionally? i.e encode state only if ephemeral rand is set on that inode?

Reply:
No, because the importer wouldn't be able to tell if it's present or not. Consider when we add more to this bufferlist in the future. We cant have arbitrary holes in the encoding.

Reply:
Ack. This reasoning is a good justification.